src/main/java/com/google/sps/data/Election.java

```diff
@@ -91,4 +71,123 @@ public Entity toEntity() {
 
     public abstract Election build();
   }
+
+  /**
+   * Creates a new Election object by extracting the properties from "electionQueryData".
+   *
+   * @param electionQueryData the JSON output of an electionQuery to the Google Civic Information
+   *     API
+   * @return the new Election object
+   */
+  public static Election fromElectionQuery(JSONObject electionQueryData) throws JSONException {
+    return Election.builder()
+        .setId(electionQueryData.getString("id"))
+        .setName(electionQueryData.getString("name"))
+        .setDate(electionQueryData.getString("electionDay"))
+        .setScope(electionQueryData.getString("ocdDivisionId"))
+        .setContests(new HashSet<Long>())
+        .setPropositions(new HashSet<Long>())
+        .build();
+  }
+
+  /**
+   * Creates a new Election object by extracting the properties from an Election object and from the
+   * properties of "voterInfoQueryData". Delegates creating Contest Entities from this election's
+   * list of contests.
+   *
+   * @param election the Election object already populated by fromElectionQuery
+   * @param datastore the Datastore to store this election's list of contests
+   * @param voterInfoQueryData the JSON output of a voterInfoQuery to the Google Civic Information
+   *     API
+   * @return the new Election object
+   */
+  public Election fromVoterInfoQuery(DatastoreService datastore, JSONObject voterInfoQueryData)
```

Comment:
Apologies if I gave you the name Election#fromVoterInfoQuery.  I think maybe "fillContestInfo()" might be more descriptive?  IMHO "voter info query" is misleading since it actually doesn't really have anything to do with info about the voter him/herself.

Reply:
Yeah, I was naming them based off of the API queries I'm getting data from.